vsphere/internal/helper/virtualmachine/virtual_machine_helper.go

```diff
@@ -206,14 +206,66 @@ func Properties(vm *object.VirtualMachine) (*mo.VirtualMachine, error) {
 	return &props, nil
 }
 
+// WaitForGuestIP waits for a virtual machine to have an IP address.
+//
+// The timeout is specified in minutes. If zero or a negative value is passed,
+// the waiter returns without error immediately.
+func WaitForGuestIP(client *govmomi.Client, vm *object.VirtualMachine, timeout int, ignoredGuestIPs []interface{}) error {
```

Comment:
What does this do differently than `WaitForGuestNet` with `wait_for_guest_net_routable` set to `false`?

Reply:
The only real difference is that it waits for the string guest.ipAddress property instead of the ArrayOfGuestNicInfo guest.net property. I thought about just adding guest.ipAddress to the list of properties that WaitForGuestNet monitors with a new case statement in the switch, but then it becomes a race between which one gets populated first in newer VMTools versions I believe. This seemed like a potential backwards compatibility problem, so I chose to make them completely independent waiters to be safe.